File: GNSDK/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnMoodgridDataPoint.java

package com.gracenote.gnsdk;

/** 
* <b>Experimental</b> 
*/ 
 
public class GnMoodgridDataPoint {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnMoodgridDataPoint(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnMoodgridDataPoint obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnMoodgridDataPoint(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public GnMoodgridDataPoint() {
    this(gnsdk_javaJNI.new_GnMoodgridDataPoint__SWIG_0(), true);
  }

  public GnMoodgridDataPoint(long x, long y) {
    this(gnsdk_javaJNI.new_GnMoodgridDataPoint__SWIG_1(x, y), true);
  }

  public void setX(long value) {
    gnsdk_javaJNI.GnMoodgridDataPoint_X_set(swigCPtr, this, value);
  }

  public long getX() {
    return gnsdk_javaJNI.GnMoodgridDataPoint_X_get(swigCPtr, this);
  }

  public void setY(long value) {
    gnsdk_javaJNI.GnMoodgridDataPoint_Y_set(swigCPtr, this, value);
  }

  public long getY() {
    return gnsdk_javaJNI.GnMoodgridDataPoint_Y_get(swigCPtr, this);
  }

}

File: GNSDK/wrappers/gnsdk_csharp/src_wrapper/GnMoodgridDataPoint.cs

namespace GracenoteSDK {

using System;
using System.Runtime.InteropServices;

/**
** GnMoodgridDataPoint
*/
public class GnMoodgridDataPoint : IDisposable {
  private HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnMoodgridDataPoint(IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new HandleRef(this, cPtr);
  }

  internal static HandleRef getCPtr(GnMoodgridDataPoint obj) {
    return (obj == null) ? new HandleRef(null, IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnMoodgridDataPoint() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnMoodgridDataPoint(swigCPtr);
        }
        swigCPtr = new HandleRef(null, IntPtr.Zero);
      }
      GC.SuppressFinalize(this);
    }
  }

  public GnMoodgridDataPoint() : this(gnsdk_csharp_marshalPINVOKE.new_GnMoodgridDataPoint__SWIG_0(), true) {
  }

  public GnMoodgridDataPoint(uint x, uint y) : this(gnsdk_csharp_marshalPINVOKE.new_GnMoodgridDataPoint__SWIG_1(x, y), true) {
  }

  public uint X {
    set {
      gnsdk_csharp_marshalPINVOKE.GnMoodgridDataPoint_X_set(swigCPtr, value);
    } 
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnMoodgridDataPoint_X_get(swigCPtr);
      return ret;
    } 
  }

  public uint Y {
    set {
      gnsdk_csharp_marshalPINVOKE.GnMoodgridDataPoint_Y_set(swigCPtr, value);
    } 
    get {
      uint ret = gnsdk_csharp_marshalPINVOKE.GnMoodgridDataPoint_Y_get(swigCPtr);
      return ret;
    } 
  }

}

}
